### PR TITLE
Improve Dashboard component: separate semantics from styles and extract subcomponents

### PR DESCRIPTION
**Pull request by gusaiani**
🎗️ Changes are covered by visual regression tests added in https://github.com/metabase/metabase/pull/17376

In preparation for #11187

Let's continue to make this file a little bit better.
As the UI is complicated, this will help crystallize the solution for making the filters sticky.

In this PR we:

* separate semantics from styles
* extract components to make the Dashboard component simpler
* add missing prop types to Dashboard component

### To Test

Nothing should have changed, just visit your dashboards and they should work as before.

### CODE REVIEW

**Comment by gusaiani** [comment]
@metabase-bot run visual tests

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+  ParametersAndCardsContainer,
+  ParametersWidgetContainer,
+} from "./Dashboard.styled";
+import DashboardHeader from "../DashboardHeader";
+import DashboardGrid from "../DashboardGrid";
+import ParametersWidget from "./ParametersWidget/ParametersWidget";
+import DashboardEmptyState from "./DashboardEmptyState/DashboardEmptyState";
+import { DashboardSidebars } from "../DashboardSidebars";
+
+import DashboardControls from "../../hoc/DashboardControls";
```
Comment (kulyk) [inline frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx:21]: Should we rearrange imports here a bit? Usually the closer the imported file is to the project root, the higher it is in the imports list. So e.g. `DashboardControls` should to under the `FullWidthContainer` I guess, imports from the same directory have to be in the bottom, etc.

**Review by kulyk — APPROVED** [review]
The code looks great, thank you @gusaiani 🏅 Couldn't spot any visual regressions in both viewing and editing modes. Have one minor suggestion on the imports order

**Comment by gusaiani** [comment]
@kulyk Thank you for your review and encouragement.

Imports reordered.

Maybe we can add linting rules to this soon 👍

**Pull request merged**